Tests/Packet++Test/Tests/SmtpTests.cpp

```diff
@@ -0,0 +1,277 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "EthLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "Packet.h"
+#include "SmtpLayer.h"
+#include "SystemUtils.h"
+#include "TcpLayer.h"
+
+PTF_TEST_CASE(SmtpParsingTests)
+{
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	// Command
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/smtpCommand.dat");
+	pcpp::Packet smtpPacket1(&rawPacket1);
+	auto *smtpLayer1 = smtpPacket1.getLayerOfType<pcpp::SmtpRequestLayer>();
+
+	PTF_ASSERT_NOT_NULL(smtpLayer1);
+	PTF_ASSERT_EQUAL(smtpLayer1->getHeaderLen(), 12);
+	PTF_ASSERT_EQUAL(smtpLayer1->getCommand(), pcpp::SmtpRequestLayer::SmtpCommand::AUTH, enumclass);
+	PTF_ASSERT_EQUAL(smtpLayer1->getCommandString(), "AUTH");
+	PTF_ASSERT_EQUAL(smtpLayer1->getCommandOption(), "LOGIN");
+	PTF_ASSERT_EQUAL(smtpLayer1->getCommandOption(false), "LOGIN");
+	PTF_ASSERT_EQUAL(smtpLayer1->toString(), "SMTP request layer, command: Authenticate client and server");
+	PTF_ASSERT_FALSE(smtpLayer1->isMultiLine());
+
+	// Response packet
+	READ_FILE_AND_CREATE_PACKET(2, "PacketExamples/smtpResponse.dat");
+	pcpp::Packet smtpPacket2(&rawPacket2);
+	auto *smtpLayer2 = smtpPacket2.getLayerOfType<pcpp::SmtpResponseLayer>();
+
+	PTF_ASSERT_NOT_NULL(smtpLayer2);
+	PTF_ASSERT_EQUAL(smtpLayer2->getHeaderLen(), 18);
+	PTF_ASSERT_EQUAL(smtpLayer2->getStatusCode(), pcpp::SmtpResponseLayer::SmtpStatusCode::AUTH_INPUT, enumclass);
+	PTF_ASSERT_EQUAL(smtpLayer2->getStatusCodeString(), "334");
+	PTF_ASSERT_EQUAL(smtpLayer2->getStatusOption(), "VXNlcm5hbWU6");
+	PTF_ASSERT_EQUAL(smtpLayer2->getStatusOption(false), "VXNlcm5hbWU6");
+	PTF_ASSERT_EQUAL(smtpLayer2->toString(), "SMTP response layer, status code: AUTH input");
+	PTF_ASSERT_FALSE(smtpLayer2->isMultiLine());
+
+	// Multiline
+	READ_FILE_AND_CREATE_PACKET(3, "PacketExamples/smtpMultiLine.dat");
+	pcpp::Packet smtpPacket3(&rawPacket3);
+	auto *smtpLayer3 = smtpPacket3.getLayerOfType<pcpp::SmtpResponseLayer>();
+
+	PTF_ASSERT_NOT_NULL(smtpLayer3);
+	PTF_ASSERT_EQUAL(smtpLayer3->getHeaderLen(), 181);
+	PTF_ASSERT_EQUAL(smtpLayer3->getStatusCode(), pcpp::SmtpResponseLayer::SmtpStatusCode::SERVICE_READY, enumclass);
+	PTF_ASSERT_EQUAL(smtpLayer3->getStatusCodeString(), "220");
+	PTF_ASSERT_EQUAL(
+		smtpLayer3->getStatusOption(),
+		"xc90.websitewelcome.com ESMTP Exim 4.69 #1 Mon, 05 Oct 2009 01:05:54 -0500 220-We do not authorize "
+		"the use of this system to transport unsolicited, 220 and/or bulk e-mail.");
```

Comment:
In case of multiline, do we want to remove the status code from all the lines?

Meaning, remove `220-` from the second and third lines?

Reply:
I remember I also thought about it but to be honest I can't remember why I don't add any support for it. Let me check the code and protocol again before merging.

Reply:
@seladb added support with f2634f4. Both FTP and SMTP follow the same structure for multiline formatting. Can you check again? If its ok for you, we can merge it.

Reply:
@seladb Updated the code for support it in both SMTP and FTP (They use same structure for it) but codecov-action currently broken because of node version (Check https://github.com/codecov/codecov-action/issues/1230 and https://github.com/codecov/codecov-action/pull/1228) so workflow is failing for now.